### ISSUE TITLE
XHarness tests broken in release/6.0

### ISSUE THREAD

**Issue by MattGal** [dotnet/arcade#8253]
- [x] This issue is blocking (CI validation of release/6.0)
- [ ] This issue is causing unreasonable pain

As seen in https://github.com/dotnet/arcade/runs/4477846178,  we've started to reliably fail the 6.0 branch Xharness tests with signing-related errors:

```
[16:08:58] dbug: Shadow directory: /tmp/Xamarin.iOS_InstallShadowPath
[16:09:15] dbug: ApplicationVerificationFailed: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.GAmo9f/extracted/System.Buffers.Tests.app : 0xe800801c (No code signature found.)
[16:09:15] dbug: error MT1006: Could not install the application '/tmp/helix/working/BE940A19/w/A62D0912/e/System.Buffers.Tests.app' on the device '00008020-001C3C593AD9002E': AMDeviceSecureInstallApplicationBundle returned: 0xe800801c.
```

This area has undergone massive refactoring since 6.0 and @premun is likely to have context (plus is on FR this week)

For now, I've used my permissions to remove the "Test Linux ..." legs from requirements for merging, but we need to deal with this ASAP. (it's fairly rare people make sweeping Helix changes there)

**Comment by premun** [comment]
What I'm seeing in that build is this error:
```
Real device target detected, application will be signed
    "/Users/helix-runner/Library/Keychains/signing-certs.keychain-db"
Warning: unable to build chain to self-signed root for signer "Apple Development: Ilya Skuratovsky (R8BUS9Y7N6)"
/tmp/helix/working/BE940A19/w/A62D0912/e/System.Buffers.Tests.app: errSecInternalComponent
```

However this seems to work on main branches (https://dev.azure.com/dnceng/public/_build/results?buildId=1509852&view=logs&j=e9980b50-2569-53d6-c997-045bbff6842b&t=c1c0cb9e-e6a3-5097-f548-5ccbf3adf664&l=44) so might be tied to old Helix SDK even though I didn't expect this to break as nothing much changed around signing itself.

The signing itself seems to work on main, will try to port the SDK to release/6.0, there shouldn't be any API changes since we branched off so it should be safe

**Comment by premun** [comment]
Linking this to https://github.com/dotnet/core-eng/issues/14888

**Comment by MattGal** [comment]
> might be tied to old Helix SDK 

@premun why not tied to the changes you made to the live machines via script right before this happened?  There weren't changes made to the Helix SDK in the 6.0 branch in this time range.  Also, can you share the script (privately pls) you ran in that case?

**Comment by premun** [comment]
> why not tied to the changes ... ?

We discussed this offline. The actual root cause is still unknown to me.

**Comment by premun** [comment]
Fixed in #8258
